daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala

```diff
@@ -1299,11 +1126,11 @@ Differences were (path, expected, actual):
    * We have to use our own PUA remapping trick if we want to be sure to preserve
    * CR in XML.
    */
-  def escape(str: String, sb: StringBuilder = new StringBuilder()): StringBuilder = {
+  def escape(s: String, sb: StringBuilder = new StringBuilder()): StringBuilder = {
```

Comment:
We do have a number of places where we call `scala.xml.Utility..escape`? I wonder if those are bugs?

I also wonder if maybe we want to do the same copy-on-write kind of thing for this function? Or maybe that's pre-optimization? Can be done in a separate PR if thats something we think we need.